docs/kb/developer-resources/updating-expiration-on-licenses.md

```diff
@@ -0,0 +1,47 @@
++++
+date = "2017-03-13T00:00:00Z"
+lastmod = "2017-03-13T00:00:00Z"
+title = "Updating Expiration on Licenses"
+weight = "999999"
+categories = [ "Knowledgebase", "Developer Resources" ]
++++
+
+A good use case for using the VendorAPI is license maintenance. In this example we want to find all licenses that are expiring soon (<5 days) in the unstable channel and extend them by a month. We are using our unstable channel to ship new features to some of our customers and want to extend their access to the channel by a month while we hammer out some bugs in this release.
+
+## Prerequisites
+
+You should already have an API Token set to Read/Write via the vendor site, and you should know the target App ID and Channel ID. For details on how to get the App ID and Channel ID programmatically, check out the document called [Finding Your API Token and App ID](/kb/developer-resources/finding-your-api-token-and-app-id). These values will not change and should be supplied as static values to your license updating script.
+
+### 1) Get all outstanding licenses
+
+First we will get all the licenses that have been issued for a specific app:
+
+    curl -X GET \
+    	 -H 'Authorization: <YOUR-API-TOKEN>' \
+    	 https://api.replicated.com/vendor/v2/app/<YOUR-APP-ID>/licenses
+
+Note, the result will contains an array of objects for all the licenses associated with this app, including all the channels.
+
+### 2) Find expiring licenses
+
+In my language of choice I am going to iterate through the returned array from step 1 and create a new array that contains all licenses where "ChannelId" fields match my Unstable Channel ID. I am then going to iterate through the new array and look for licenses with an "ExpireDate" that is within the next five days and pop those from this array.
```

Comment:
my should read as your as your empowering the reader, also read up on writing in the active voice, so for example the first sentence would be

In your language of choice iterate through the JSON array from step 1 and filter them to contains only licenses where the "ChannelId" match your beta channel.